zap/src/test/java/org/zaproxy/zap/authentication/FormBasedAuthenticationMethodTypeUnitTest.java

```diff
@@ -0,0 +1,248 @@
+ * limitations under the License.
+ */
+package org.zaproxy.zap.authentication;
+
+import static fi.iki.elonen.NanoHTTPD.newFixedLengthResponse;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import fi.iki.elonen.NanoHTTPD.IHTTPSession;
+import fi.iki.elonen.NanoHTTPD.Response;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.authentication.AuthenticationMethod.AuthCheckingStrategy;
+import org.zaproxy.zap.authentication.AuthenticationMethod.AuthPollFrequencyUnits;
+import org.zaproxy.zap.authentication.PostBasedAuthenticationMethodType.PostBasedAuthenticationMethod;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.session.SessionManagementMethod;
+import org.zaproxy.zap.session.WebSession;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+import org.zaproxy.zap.testutils.TestUtils;
+import org.zaproxy.zap.users.User;
+import org.zaproxy.zap.utils.I18N;
+
+@ExtendWith(MockitoExtension.class)
+public class FormBasedAuthenticationMethodTypeUnitTest extends TestUtils {
+
+    private static final String LOGGED_IN_INDICATOR = "logged in";
+    private static final String LOGGED_IN_BODY =
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit. "
+                    + "Pellentesque auctor nulla id turpis placerat vulputate."
+                    + LOGGED_IN_INDICATOR
+                    + " Proin tempor bibendum eros rutrum. ";
+
+    private AuthenticationMethod method;
+    private FormBasedAuthenticationMethodType type;
+    private static Context mockedContext;
+    private static SessionManagementMethod mockedSessionManagementMethod;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+
+        Constant.getInstance();
+        I18N i18n = Mockito.mock(I18N.class, withSettings().lenient());
+        given(i18n.getString(anyString())).willReturn("");
+        given(i18n.getString(anyString(), any())).willReturn("");
+        given(i18n.getLocal()).willReturn(Locale.getDefault());
+        Constant.messages = i18n;
```

Comment:
Worth exposing a method, e.g. `WithConfigsTest#setupConstant()` instead of duplicating here.